airflow/executors/celery_executor.py

```diff
@@ -543,20 +539,27 @@ def __init__(self, sync_parallelism=None):
         super().__init__()
         self._sync_parallelism = sync_parallelism
 
+    def _tasks_list_to_task_ids(self, async_tasks) -> Set[str]:
+        return {a.task_id for a in async_tasks}
+
     def get_many(self, async_results) -> Mapping[str, EventBufferValueType]:
         """Gets status for many Celery tasks using the best method available."""
         if isinstance(app.backend, BaseKeyValueStoreBackend):
             result = self._get_many_from_kv_backend(async_results)
-            return result
-        if isinstance(app.backend, DatabaseBackend):
+        elif isinstance(app.backend, DatabaseBackend):
             result = self._get_many_from_db_backend(async_results)
-            return result
-        result = self._get_many_using_multiprocessing(async_results)
-        self.log.debug("Fetched %d states for %d task", len(result), len(async_results))
+        else:
+            async_results = list(async_results) if isinstance(async_results, map) else async_results
```

Comment:
That's probably the most intuitive fix, but in the case of the recommended backends (i.e. `BaseKeyValueStoreBackend` or `DatabaseBackend`, we'd be converting it to a list and then immediately creating a set based off that list instead of just creating the set based on the map object. Do you think that could have a potential performance hit if a lot of tasks are running?

Reply:
@ashb thoughts?

Reply:
@RNHTTR I think most of the cases are not large enough for this to make much of a difference - the only map case is in code that runs once every 30s, and is not performance _critical_, so for ease of understanding, lets change the one usage instead of complicating this case.

Reply:
Sounds good. PR updated. Any resources you recommend on diving into some of airflow's trickier topics? I can't seem to find much on task adoption for example. We run managed airflow, so a lot of airflow's challenging aspects are abstracted away.

Reply:
Task Adoption is when running a scheduler, if it dies another scheduler is able to "adopt" tasks started by that Scheduler/executor -- this currently works for Celery, Kubernetes (and CeleryKubernete) executor

Code:

https://github.com/apache/airflow/blob/4d1b2e985492894e6064235688cf8f381b1e8858/airflow/jobs/scheduler_job.py#L1825

Reply:
Thanks Kaxil that's helpful. Do you think it'd be a good idea to put together a GLOSSARY.rst or similar to capture some of these more nuanced concepts?

Reply:
oh yeah probably